yew/src/virtual_dom/vtag.rs

```diff
@@ -1480,3 +1525,204 @@ mod layout_tests {
         diff_layouts(vec![layout1, layout2, layout3, layout4]);
     }
 }
+
+#[cfg(all(test, feature = "web_sys"))]
+mod benchmarks {
+    use super::{Patch, VTag};
+    use easybench_wasm::bench_env_limit;
+    use indexmap::map::IndexMap;
+    use std::borrow::Cow;
+    use std::collections::HashMap;
+
+    #[cfg(feature = "wasm_test")]
+    use wasm_bindgen_test::{wasm_bindgen_test as test, wasm_bindgen_test_configure};
+
+    #[cfg(feature = "wasm_test")]
+    wasm_bindgen_test_configure!(run_in_browser);
+
+    // In seconds
+    const BENCHMARK_DURATION: f64 = 1.0;
+
+    fn diff_attributes_hashmap<'a>(
+        new: &'a HashMap<&'static str, Cow<'static, str>>,
+        old: &'a HashMap<&'static str, Cow<'static, str>>,
+    ) -> Vec<Patch<&'static str, &'a str>> {
+        // Only change what is necessary.
+        let to_add_or_replace = new
+            .iter()
+            .filter_map(move |(key, value)| match old.get(key) {
+                None => Some(Patch::Add(&**key, &**value)),
+                Some(ancestor_value) if value != ancestor_value => {
+                    Some(Patch::Replace(&**key, &**value))
+                }
+                _ => None,
+            });
+        let to_remove = old
+            .keys()
+            .filter(move |key| !new.contains_key(&**key))
+            .map(|key| Patch::Remove(&**key));
+
+        to_add_or_replace.chain(to_remove).collect()
+    }
+
+    fn diff_attributes_indexmap<'a>(
+        new: &'a IndexMap<&'static str, Cow<'static, str>>,
+        old: &'a IndexMap<&'static str, Cow<'static, str>>,
+    ) -> Vec<Patch<&'static str, &'a str>> {
+        // Only change what is necessary.
```

Comment:
The benefit of index map is that ordering is consistent so this should be implemented similarly to the vector approach where you iterate over `new` and `old` together to optimize for the common case.

Reply:
Yes, it can, but an `IndexedMap` is still more expensive to construct than a `Vec`. See my proposal above.